src/cli/Create.cpp

```diff
@@ -70,9 +71,8 @@ Create::Create()
  *
  * @return EXIT_SUCCESS on success, or EXIT_FAILURE on failure
  */
-int Create::execute(const QStringList& arguments)
+int Create::createDataBase(const QSharedPointer<QCommandLineParser>& parser, bool importDatabase)
```

Comment:
@Colfenor I feel like we could be able to remove the `importDatabase` argument. We could just create an empty database in the case of a `Create`, and import the xml in the case of `Import`, and then pass to db to this function, which would set the keys. The function would be named `setEncryptionKeys` though. What do you think?

Reply:
I would say it basically breaks down of how the initializing of `QString databaseFilename` is different in the 2 commands. In Create it is the first parsed argument vs in Import it is the second argument because the first one would be the xml filePath. 

Now the `setEncryptionKeys` function would need then the `databaseFilename` string and the database object as arguments both are then needed in the final `dataBase.saveAs(.....)` attempt :)

Reply:
The other option would be to save the database after calling `setEncryptionKeys`. This would lead to a bit of code duplication, but we have that database saving code in a lot of the CLI commands anyway. At least all of the code related to handling the encryption options would be centralized.

Reply:
Hm I checked on this option, it's in my opinion not ideal. 
Just to make sure we are on the same page, would your suggested function `setEncryptionKeys` return a `QSharedPointer<CompositeKey>` ?

If so we still have the code of the `decryptionTime` duplicated which is determined from the parser object and later on set on the actual new database object, which again would be outside of `setEncryptionKeys` in this design choice. 

But yeah the **keyfile & password** setting code would be centralized.

Reply:
> would your suggested function setEncryptionKeys return a QSharedPointer<CompositeKey>

It could return a `QSharedPointer<Database>` instead? This way we could set the decryption time also in the function. I guess `setEncryptionKeys` is not a proper name. In the future, we would also handle options like setting the database name and the database description in that function, for example. Maybe `createDatabaseFromOptions`?

Reply:
> It could return a `QSharedPointer<Database>` instead? 

Yeah sure :) similar to the `unlockDatabase` function in `Utils` ! So for me the question still remains what should happen in the Create or Import `::execute` function. Currently from the latest commit it calls `createDatabase`where all the logic is contained.

If some logic like the `databaseFilename` creation and finally receiving the `db` object + checking/saving it, should be for instance in `::execute` we still have the parser duplicated which we would need in both functions.

Reply:
Not sure what you mean by `have the parser variable duplicated`, but both commands would access the parser variable in their `execute` function for saving the database. I think this is inevitable. I don't see a problem with both the execute functions and the `createDatabaseFromOptions` handling the `parser` variable.

Maybe we could rename it `initializeDatabaseFromOptions`, since the actual creation (i.e. saving or importing) will be done after calling that function, in the `execute` of the commands.

Reply:
> Not sure what you mean by `have the parser variable duplicated`, but both commands would access the parser variable in their `execute` function for saving the database. I think this is inevitable. I don't see a problem with both the execute functions and the `createDatabaseFromOptions` handling the `parser` variable.

Yes exactly that's what I mean, ok then we have it like that.
I just committed and hopefully the changes are still in the scope of being readable. 
The create & import options man page is also merged now and the import tests are updated with decryption time.

Thanks a lot for your patience & review so far :) !